diff --git a/node_modules/react-native-blur/android/src/main/java/com/cmcewen/blurview/BlurringView.java b/node_modules/react-native-blur/android/src/main/java/com/cmcewen/blurview/BlurringView.java
index b1fc48a..8a3f82a 100644
--- a/node_modules/react-native-blur/android/src/main/java/com/cmcewen/blurview/BlurringView.java
+++ b/node_modules/react-native-blur/android/src/main/java/com/cmcewen/blurview/BlurringView.java
@@ -7,10 +7,10 @@ import android.graphics.Bitmap;
 import android.graphics.Canvas;
 import android.graphics.Color;
 import android.graphics.drawable.ColorDrawable;
-import android.support.v8.renderscript.Allocation;
-import android.support.v8.renderscript.Element;
-import android.support.v8.renderscript.RenderScript;
-import android.support.v8.renderscript.ScriptIntrinsicBlur;
+import androidx.renderscript.Allocation;
+import androidx.renderscript.Element;
+import androidx.renderscript.RenderScript;
+import androidx.renderscript.ScriptIntrinsicBlur;
 import android.util.AttributeSet;
 import android.view.View;
 
