diff --git a/node_modules/react-native-gesture-handler/AndroidNativeExample/app/src/main/java/com/swmansion/gesturehandler/example/MainActivity.java b/node_modules/react-native-gesture-handler/AndroidNativeExample/app/src/main/java/com/swmansion/gesturehandler/example/MainActivity.java
index a293839..5b5054c 100644
--- a/node_modules/react-native-gesture-handler/AndroidNativeExample/app/src/main/java/com/swmansion/gesturehandler/example/MainActivity.java
+++ b/node_modules/react-native-gesture-handler/AndroidNativeExample/app/src/main/java/com/swmansion/gesturehandler/example/MainActivity.java
@@ -1,7 +1,7 @@
 package com.swmansion.gesturehandler.example;
 
 import android.os.Bundle;
-import android.support.v7.app.AppCompatActivity;
+import androidx.appcompat.app.AppCompatActivity;
 import android.view.MotionEvent;
 import android.view.View;
 import android.widget.Button;
diff --git a/node_modules/react-native-gesture-handler/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerEvent.java b/node_modules/react-native-gesture-handler/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerEvent.java
index 85dca41..43f0e6a 100644
--- a/node_modules/react-native-gesture-handler/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerEvent.java
+++ b/node_modules/react-native-gesture-handler/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerEvent.java
@@ -1,6 +1,6 @@
 package com.swmansion.gesturehandler.react;
 
-import android.support.v4.util.Pools;
+import androidx.core.util.Pools;
 
 import com.facebook.react.bridge.Arguments;
 import com.facebook.react.bridge.WritableMap;
diff --git a/node_modules/react-native-gesture-handler/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerStateChangeEvent.java b/node_modules/react-native-gesture-handler/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerStateChangeEvent.java
index 5912a45..8559366 100644
--- a/node_modules/react-native-gesture-handler/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerStateChangeEvent.java
+++ b/node_modules/react-native-gesture-handler/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerStateChangeEvent.java
@@ -1,6 +1,6 @@
 package com.swmansion.gesturehandler.react;
 
-import android.support.v4.util.Pools;
+import androidx.core.util.Pools;
 
 import com.facebook.react.bridge.Arguments;
 import com.facebook.react.bridge.WritableMap;
diff --git a/node_modules/react-native-gesture-handler/e2e/android/app/src/androidTest/java/com/swmansion/gesturehandler/react/example/DetoxTest.java b/node_modules/react-native-gesture-handler/e2e/android/app/src/androidTest/java/com/swmansion/gesturehandler/react/example/DetoxTest.java
index 5709e38..9bbc714 100644
--- a/node_modules/react-native-gesture-handler/e2e/android/app/src/androidTest/java/com/swmansion/gesturehandler/react/example/DetoxTest.java
+++ b/node_modules/react-native-gesture-handler/e2e/android/app/src/androidTest/java/com/swmansion/gesturehandler/react/example/DetoxTest.java
@@ -1,8 +1,8 @@
 package com.swmansion.gesturehandler.react.example;
 
-import android.support.test.filters.LargeTest;
-import android.support.test.rule.ActivityTestRule;
-import android.support.test.runner.AndroidJUnit4;
+import androidx.test.filters.LargeTest;
+import androidx.test.rule.ActivityTestRule;
+import androidx.test.runner.AndroidJUnit4;
 
 import com.wix.detox.Detox;
 
