Index: src/store/FileStore.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/store/FileStore.js	(revision )
+++ src/store/FileStore.js	(revision )
@@ -32,7 +32,7 @@
         .c("download-request", {xmlns: NS})
         .c("id", {}).t(url);
 
-      console.log("WAITING FOR IQ RESPONSE", iq);
+      console.log("WAITING FOR IQ RESPONSE");
       let data = await this.xmpp.sendIQ(iq);
       data = data.download;
       assert(data, "data should be defined");
